common/replication/block_fetcher.go

```diff
@@ -509,9 +527,14 @@ func (bf *BlockFetcher) PullBlock(seq uint64) *common.Block {
 
 		if block != nil {
 			bf.Logger.Debugf("Got block %d from %s", block.Header.Number, bf.currentEndpoint.Endpoint)
-			if err := bf.VerifyBlock(block.Header, block.Metadata); err != nil {
-				bf.Logger.Warnf("Failed verifying signature on block %d received from %s: %v", block.Header.Number, bf.currentEndpoint.Endpoint, err)
-				return nil
+			/* In case `VerifyBlock` Function is not set, genesis block is being pulled, so no need to verify it.*/
+			if !bf.setupVerifierInProgress {
+				if err := bf.VerifyBlock(block.Header, block.Metadata); err != nil {
+					if block.Header.Number != 0 {
```

Comment:
In which scenario we have set up the verifier but we try to pull block 0? How is that possible?

Reply:
my bad. This check was added for testing earlier. I will remove it.